protocol-specs/generators/multisignature/index.js

```diff
@@ -0,0 +1,433 @@
+	},
+	mandatoryOne: {
+		passphrase:
+			'trim elegant oven term access apple obtain error grain excite lawn neck',
+		privateKey:
+			'8a138c0dd8efe597c8b9c519af69e9821bd1e769cf0fb3490e22209e9cabfb8df1b9f4ee71b5d5857d3b346d441ca967f27870ebee88569db364fd13e28adba3',
+		publicKey:
+			'f1b9f4ee71b5d5857d3b346d441ca967f27870ebee88569db364fd13e28adba3',
+		address: '10481548956627905381L',
+	},
+	mandatoryTow: {
+		passphrase:
+			'desk deposit crumble farm tip cluster goose exotic dignity flee bring traffic',
+		privateKey:
+			'ddc8e19d6697d6e5c1dacf6576a7169752810999918212afe14d3978b354f8aa4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
+		publicKey:
+			'4a67646a446313db964c39370359845c52fce9225a3929770ef41448c258fd39',
+		address: '3372320078773139180L',
+	},
+	optionalOne: {
+		passphrase:
+			'sugar object slender confirm clock peanut auto spice carbon knife increase estate',
+		privateKey:
+			'69aa94ea7ade3b7b08e277b18c1a590b2306ce5973ae8462b0b85122b180e89c57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
+		publicKey:
+			'57df5c3811961939f8dcfa858c6eaefebfaa4de942f7e703bf88127e0ee9cca4',
+		address: '7745870967079479156L',
+	},
+	optionalTwo: {
+		passphrase:
+			'faculty inspire crouch quit sorry vague hard ski scrap jaguar garment limb',
+		privateKey:
+			'ffed38380998a90a2af9501f10182bc2a07922448ab383575b1e34aeddfa5482fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
+		publicKey:
+			'fa406b6952d377f0278920e3eb8da919e4cf5c68b02eeba5d8b3334fdc0369b6',
+		address: '7086965981385941478L',
+	},
+};
+
+const sortKeysDescending = publicKeys =>
+	publicKeys.sort((publicKeyA, publicKeyB) => {
+		// eslint-disable-next-line no-undef, new-cap
+		if (BigInt(`0x${publicKeyA}`) > BigInt(`0x${publicKeyB}`)) return 1;
+		// eslint-disable-next-line no-undef, new-cap
+		if (BigInt(`0x${publicKeyA}`) < BigInt(`0x${publicKeyB}`)) return -1;
+		return 0;
+	});
+
+const getId = transactionBytes => {
+	const transactionHash = hash(transactionBytes);
+	const bufferFromFirstEntriesReversed = getFirstEightBytesReversed(
+		transactionHash,
+	);
+	const transactionId = bufferToIntAsString(bufferFromFirstEntriesReversed);
+
+	return transactionId;
+};
+
+const assetToBytes = tx => {
+	const { mandatoryKeys, optionalKeys, numberOfSignatures } = tx.asset;
+	const mandatoryKeysBuffer = Buffer.from(mandatoryKeys.join(''), 'hex');
+	const optionalKeysBuffer = Buffer.from(optionalKeys.join(''), 'hex');
+
+	const assetBuffer = Buffer.concat([
+		intToBuffer(mandatoryKeys.length, 1),
+		mandatoryKeysBuffer,
+		intToBuffer(optionalKeys.length, 1),
+		optionalKeysBuffer,
+		intToBuffer(numberOfSignatures, 1),
+	]);
+	return assetBuffer;
+};
+
+const createSignatureObject = (txBuffer, account) => ({
+	signature: signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
+		account.passphrase,
+	),
+});
+
+const serializeBasicProperties = tx => {
+	const transactionTimestamp = Buffer.alloc(4);
```

Comment:
are we gonna continue using timestamp?

Reply:
In the open PR for multi-signatures it is still there. We can update later if it's being removed.